dcp/api/job.py

```diff
@@ -1,13 +1,128 @@
 import pythonmonkey as pm
+import cloudpickle
+import dill
+import asyncio
+from ..js import utils
+from .. import dry
+from .job_serializers import Serializers
+from .job_env import Env
+from .job_modules import Modules
+from .job_fs import JobFS
+from collections.abc import Iterator
+from types import FunctionType
+import urllib.parse
+from .pyodide_work_function import work_function_string
+
 
 def job_maker(super_class):
     class Job(super_class):
-        #def exec(self, *args, **kwargs):
-        #    # TODO change behaviour to match spec
-        #    #print("overidden by Job hook")
+        def __init__(self, *args, **kwargs):
+            compute_for_js = pm.eval("globalThis.dcp.compute.for")
+            job_js = dry.aio.blockify(compute_for_js)(*args, **kwargs)
+            job_js.worktime = "pyodide"
+            super().__init__(job_js)
+
+            work_function = None
+            work_function_candidates = [arg for arg in args if isinstance(arg, FunctionType)]
+            if len(work_function_candidates):
+                 work_function = work_function_candidates[0]
+
+            self._wrapper_set_attribute("_work_function", work_function)
+            self._wrapper_set_attribute("_serializers_instance", Serializers())
+            self._wrapper_set_attribute("_env_instance", Env())
+            self._wrapper_set_attribute("_modules_instance", Modules())
+            self._wrapper_set_attribute("fs", JobFS())
+            self._wrapper_set_attribute("exec_called", False)
```

Comment:
should `exec_called` be underscored? up to you, the concept of "underscore" signals at the very least that its an implementation detail

Reply:
done